File: app/src/main/java/com/openhands/tvgamerefund/data/models/Game.kt
package com.openhands.tvgamerefund.data.models

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey
import java.util.Date

/**
 * Représente un jeu télévisé
 */
@Entity(
    tableName = "games",
    foreignKeys = [
        ForeignKey(
            entity = Show::class,
            parentColumns = ["id"],
            childColumns = ["showId"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [Index("showId")]
)
data class Game(
    @PrimaryKey
    val id: String,
    val showId: String,
    val title: String,
    val description: String,
    val type: GameType,
    val startDate: Date?,
    val endDate: Date?,
    val rules: String,
    val imageUrl: String?,
    val participationMethod: String,
    val reimbursementMethod: String,
    val reimbursementDeadline: Int, // Délai en jours pour demander remboursement
    val cost: Double = 0.0,         // Coût de participation
    val phoneNumber: String = "",   // Numéro pour participer
    val refundAddress: String = "", // Adresse pour remboursement
    val isLiked: Boolean = false,   // Favori
    val createdAt: Date = Date(),
    val updatedAt: Date = Date(),
    // Champs temporaires pour la compatibilité avec le code existant
    val showName: String = "",
    val channel: String = "",
    val airDate: Date? = null,
    val gameType: String = "",
    val refundDeadline: Int = 0
)

File: backend/models/Game.js
const mongoose = require('mongoose');

const gameSchema = new mongoose.Schema({
  id: { type: String, required: true, unique: true },
  showId: { type: String, required: true },
  title: { type: String, required: true },
  description: { type: String, default: '' },
  type: { 
    type: String, 
    enum: ['SMS', 'PHONE_CALL', 'MIXED'], 
    default: 'SMS' 
  },
  startDate: { type: Date, default: Date.now },
  endDate: { type: Date, default: null },
  rules: { type: String, default: '' },
  imageUrl: { type: String, default: null },
  participationMethod: { type: String, default: '' },
  reimbursementMethod: { type: String, default: '' },
  reimbursementDeadline: { type: Number, default: 60 },
  cost: { type: Number, default: 0.99 },
  phoneNumber: { type: String, default: '' },
  refundAddress: { type: String, default: '' },
  channel: { type: String, default: 'TF1' },
  tmdbId: { type: Number, default: null },
  createdAt: { type: Date, default: Date.now },
  updatedAt: { type: Date, default: Date.now }
});

// Index pour les recherches
gameSchema.index({ title: 'text', description: 'text' });
gameSchema.index({ showId: 1 });
gameSchema.index({ channel: 1 });

const Game = mongoose.model('Game', gameSchema);

module.exports = Game;